### ISSUE TITLE
Should we have a "train_native.py"?

### ISSUE THREAD

**Issue by 6DammK9** [kohya-ss/sd-scripts#1947]
*This issus is majorly focusing on code structure*.

Currently I'm working on porting [resume from assigned epoch / iter](https://github.com/kohya-ss/sd-scripts/pull/1359/) and [bundled validation loss](https://github.com/kohya-ss/sd-scripts/pull/1899) to `sdxl_train.py`, to enable massive "native full finetune" in my SDXL base model.
I'm currently working in `sd3` "WIP branch".

I have found that many (basic / fundamental / general) features are implemented in `class NetworkTrainer`, which is no accessible in `*_train.py`.

Meanwhile ARB / latent cache related configurations, even the implementation itself,  I have made my own scalable version of [prepare_buckets_latents.py](https://github.com/6DammK9/nai-anime-pure-negative-prompt/blob/main/ch06/sd-scripts-runtime/prepare_buckets_latents_v2.py), and made a [huge latent dataset](https://huggingface.co/datasets/6DammK9/danbooru2024-latents-sdxl-1ktar), realizing that I am close to have invalidate configuration because of the *inconsistent magic numbers* in `verify_bucket_reso_steps`.
Moreover, the `super()` will amplify this issue if we use downsteram applications / extensions, such as LyCORIS's ["full bypass"](https://github.com/KohakuBlueleaf/LyCORIS/blob/0538b17fe30acd27900a8efe829fdfb6f5016430/lycoris/modules/full.py#L69), which may hide the stack trace and the actual code dependency.

Examining the newer coding structures shared in `train_*.py`, maybe we should have a `train_naive.py` to unify the implementation diifference spreaded across arch specific `*_train.py`. 

Any actions able to mitigate this risk will be greatly appreciated.

PS: `accelerator.skip_first_batches` in `sdxl_train.py` "soon".
